src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,220 @@
+      };
+
+      private double childPughScoreFor(Individual individual)
+      {
+         return individual.OriginData.DiseaseStateParameters.FindByName(CHILD_PUGH_SCORE).Value;
+      }
+
+      public bool ApplyForPopulationTo(Individual individual)
+      {
+         throw new NotImplementedException();
+      }
+
+      public Individual CreateBaseIndividualForPopulation(Individual originalIndividual)
+      {
+         throw new NotImplementedException();
+      }
+
+      public void ResetParametersAfterPopulationIteration(Individual individual)
+      {
+         throw new NotImplementedException();
+      }
+
+      private void updateParameterValue(IParameter parameter, double value, bool useAsFactor)
+      {
+         var valueToUse = useAsFactor ? parameter.Value * value : parameter.Value;
+         parameter.Value = valueToUse;
+      }
+
+      private void updateParameter(IParameter parameter, double value, bool useAsFactor)
+      {
+         var valueToUse = useAsFactor ? parameter.Value * value : parameter.Value;
+         updateValueOriginsFor(parameter);
+         if (parameter is IDistributedParameter distributedParameter)
+         {
+            distributedParameter.ScaleDistributionBasedOn(valueToUse / distributedParameter.Value);
+            return;
+         }
+
+         //We are using a formula, we override with a constant
+         if (parameter.Formula.IsExplicit())
+         {
+            parameter.Formula = _formulaFactory.ConstantFormula(valueToUse, parameter.Dimension);
+            //Make sure the formula is indeed used in case the value was overwritten before as fixed value
+            parameter.IsFixedValue = false;
+            return;
+         }
+
+         //constant formula
+         updateParameterValue(parameter, value, useAsFactor);
+         parameter.DefaultValue = valueToUse;
+         parameter.IsFixedValue = false;
+      }
+
+      private void updateValueOriginsFor(IParameter parameter)
+      {
+         parameter.ValueOrigin.UpdateAllFrom(_valueOriginRepository.FindBy(HI_VALUE_ORIGIN_ID));
+         //Make sure we mark this parameter as changed by create individual. It might already be the case but in that case, it does not change anything
+         parameter.IsChangedByCreateIndividual = true;
+      }
+
+      public void Validate(OriginData originData)
+      {
+         var (valid, error) = IsValid(originData);
+         if (valid)
+            return;
+
+         throw new OSPSuiteException(error);
+      }
+
+      public (bool isValid, string error) IsValid(OriginData originData)
+      {
+         var ageInYears = _ageDimension.BaseUnitValueToUnitValue(_ageDimension.Unit(Years), originData.Age.Value);
+         if (ageInYears >= 18)
+            return (true, string.Empty);
+
+         return (false, PKSimConstants.Error.HIOnlyAvailableForAdult);
+      }
+
+      public void ApplyTo(IndividualMolecule individualMolecule)
+      {
```

Comment:
Here in fact we have some parameters
![image](https://user-images.githubusercontent.com/1041237/235110266-270358a8-3c8d-44b0-85a6-ec1c6d9d3b65.png)

Reply:
not sure what's the best way to code this.